packages/vite/src/node/plugins/resolve.ts

```diff
@@ -894,18 +898,18 @@ export async function tryOptimizedResolve(
     // this narrows the need to do a full resolve
     if (!pkgPath.endsWith(id)) continue
 
-    // lazily initialize resolvedSrc
-    if (resolvedSrc == null) {
-      try {
-        // this may throw errors if unable to resolve, e.g. aliased id
-        resolvedSrc = normalizePath(resolveFrom(id, path.dirname(importer)))
-      } catch {
-        // this is best-effort only so swallow errors
-        break
-      }
+    try {
+      // compute resolvedSrc lazily because resolveFrom is expensive
+      // this may throw errors if unable to resolve, e.g. aliased id
+      resolvedSrc = normalizePath(resolveFrom(id, path.dirname(importer)))
+    } catch {
+      // this is best-effort only so swallow errors
     }
+    // if we can't compute resolvedSrc, return early
+    // no need to try again in next iteration
+    if (!resolvedSrc) return
```

Comment:
Should we still keep the `if (resolvedSrc == null) {` wrapper? Otherwise if it successfully resolves, it will still continuously resolve in the next loop and so on.